openpype/modules/kitsu/plugins/publish/integrate_kitsu_review.py

```diff
@@ -29,10 +30,46 @@ def process(self, instance):
             # Skip if not tagged as review
             if "kitsureview" not in representation.get("tags", []):
                 continue
+
+            files_idx = representation.get("files")
+
+            custom_frames = instance.data.get("customFrames", [])
+            keep_frame_index = instance.data.get("keepFrameIndex", False)
+            frame_start = instance.data["frameStart"]
+            frame_end = instance.data["frameEnd"]
+
+            # If only one frame force a list
+            if not isinstance(files_idx, list):
+                files_idx = [files_idx]
+
+            extension = representation.get("ext")
+            if not extension:
+                self.log.warning("No extension found in representation.")
+                raise IndexError
+
             review_path = representation.get("published_path")
+            if not review_path:
+                self.log.warning("No publish path found in representation.")
+                raise IndexError
             self.log.debug("Found review at: {}".format(review_path))
-            gazu.task.add_preview(
-                task_id, comment_id, review_path,
-                normalize_movie=getattr(self, 'normalize', True)
-            )
+
+            if "burnin" in representation.get("tags", []):
+                enumerate_files = zip(list(range(frame_start, frame_end+1)), files_idx)
```

Comment:
Multiple points:
* Why using the zip method since you only use the first item of the tuple (the index)
* If you remove the `zip` method, `enumerate_files` should be named `frame_indexes` or `file_indexes`
* Not fan of the abbreviation of `index` and `indexes` to `idx` and `idxs` since it's close and similar to `id` / `ids`. `id` is for **identifier**, which is not the same as **index**. This could lead to confusions


So in my opinion the code should look like this:

```python
file_indexes = list(range(frame_start, frame_end+1))
if custom_frames:
    file_indexes = custom_frames
filenames = ["{:04d}.{}".format(index, extension) for index in file_indexes]

for filename in filenames:
    image_filepath = self._rename_output_filepath(review_path, extension, filename)
    self.log.info(image_filepath)
    
    gazu.task.add_preview(
        task_id, comment_id, image_filepath, normalize_movie=getattr(self, 'normalize', True)
    )
```

Reply:
Got it :)

Reply:
I'll just name the filenames - > filenames_indexes because they are like "0001.p,g", it's not the full file name, more like indicative representation files names

Reply:
I should just keep the burn_in condition, since the filenames in burnIns and in the final published folder are not named the same. It's a security to make sure to uplaod the correct files